plugins/sla-email/src/main/java/com/thinkbiganalytics/metadata/sla/SlaEmailService.java

```diff
@@ -51,12 +54,23 @@
      * @param body    the email body
      */
     public void sendMail(String to, String subject, String body) {
-        SimpleMailMessage message = new SimpleMailMessage();
-        message.setFrom(emailConfiguration.getFrom());
-        message.setTo(to);
-        message.setSubject(subject);
-        message.setText(body);
-        mailSender.send(message);
+
+        try {
+            if(testConnection()) {
+                MimeMessage message = mailSender.createMimeMessage();
+                String fromAddress = StringUtils.defaultIfBlank(emailConfiguration.getFrom(),emailConfiguration.getUsername());
+                message.setFrom(new InternetAddress(fromAddress));
+                message.setRecipients(Message.RecipientType.TO, InternetAddress.parse(to));
+                message.setSubject(subject);
+                message.setText(body);
+                mailSender.send(message);
+                log.debug("Email send to " + to);
+            }
+        } catch (MessagingException ex) {
+            log.error("Exception while sending mail :" + ex.getMessage());
+            throw new RuntimeException(ex);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)